.github/workflows/changesets-status.yml

```diff
@@ -0,0 +1,31 @@
+name: Changesets
+
+on: [pull_request]
+
+jobs:
+  status:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: ".nvmrc"
+      - name: Cache node modules
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-qa-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-qa-${{ env.cache-name }}-
+            ${{ runner.os }}-qa-
+            ${{ runner.os }}-
+      - name: Install deps
+        run: npm ci
```

Comment:
This is probably overkill to install all dependencies rather than only `changesets`, it would most likely save a considerable amount of time.

A convention I have on my personal repositories is to store workflow-specific dependencies under `.github/requirements/<workflow-name>` then I install them in the workflow so only what's needed is installed

Reply:
hmm make sense, ofc we can't install just one library (npm will install everything anyway), but... probably we can use npx here